src/pages/events/index.tsx

```diff
@@ -0,0 +1,76 @@
+import React, { useEffect, useState } from "react"
+import EventCard from "@/components/events/event-card"
+import { useSupabaseClient } from "@supabase/auth-helpers-react"
+import { Event } from "@/types/database-types"
+import { getEventsWithAttendees } from "@/services/event.service"
+import { Database } from "@/types/database"
+import { MagnifyingGlassCircleIcon } from "@heroicons/react/24/outline"
+import Spinner from "@/components/utils/spinner"
+
+const Index = () => {
+  const supabaseClient = useSupabaseClient<Database>()
+  const [events, setEvents] = useState<Event[]>()
+  const [showSpinner, setShowSpinner] = useState<boolean>(true)
+  const [errorText, setErrorText] = useState("")
+  const fetchEvents = () => {
+    setShowSpinner(true)
+    const events = getEventsWithAttendees(supabaseClient)
+    events
+      .then((res) => {
+        if (res.error) {
+          setErrorText("Failed to load events, try refreshing the page.")
```

Comment:
Users on mobile can't refresh the page so I would just keep it at failed to load events